### DISCUSSION TITLE
RustDesk Privacy - Why is Telemetry forced on the Android App?  Are there other concerns for the desktop app?

### DISCUSSION THREAD

**Discussion by distinctjuggle** [rustdesk/rustdesk#10095] [Q&A] [answered]
I was looking at giving RustDesk a try when I noticed that the F-Droid client lists tracking as an anti-feature.  

https://f-droid.org/en/packages/com.carriez.flutter_hbb/

Upon further investigation, I found this thread:

https://forum.f-droid.org/t/rustdesk-tracking-features/27403

Is there no way to disable this telemetry?  Even if it's not near as egregious or serious as other things out there, it'd be better for the project as a whole to not have this issue.  

___

I also wanted to clarify exactly what is going on with these three concerns.  If I'm reading this correctly, the second and third point are both connecting to RustDesk's public servers, right?  This is NOT connecting to self-hosted servers only, or no servers only if no servers are configured, right?  I'd of course be fine with these connections if they're reaching out to my self-hosted server (assuming the servers have no similar concerns), but that doesn't seem to be the case.  

Any way that this could be addressed?  As I said above, it's not near as egregious or concerning as a lot of things out there, but the mere ability to remove the, "tracking" anti-feature from the F-Droid page would be a much better look for RustDesk as a whole.  Thanks.  

___

Oh, and one final thought - are there any similar concerns for the desktop clients/servers/software?

**Comment by rustdesk** [comment]
@basilgello

**Reply by basilgello** [reply, marked as answer]
All that is needed to resolve the issue is to introduce GUI toggle like https://github.com/rustdesk/rustdesk/pull/10165/commits/ac26ef01d32470f9746e735b1a603f508a6062f8 for https://github.com/rustdesk/rustdesk/blob/master/src/common.rs#L620 and https://github.com/rustdesk/rustdesk/blob/master/src/common.rs#L497

Also I am not sure if https://github.com/rustdesk/rustdesk/pull/10165/commits/ac26ef01d32470f9746e735b1a603f508a6062f8#diff-4b2dbedc1ec8768998fccc50ee860817ca91b52c60d7d66c766fec2eb6fa8a03R86 leaks info or not
